File: Backend/SOP/SOP/Archive/Entities/Archive_Item.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
namespace SOP.Archive.Entities
{
    public class Archive_Item
    {
        [Key]
        public int Id { get; set; }

        [Column(TypeName = "datetime")]
        public DateTime DeleteTime { get; set; }

        [Column(TypeName = "tinyint")]
        public int RoomId { get; set; }

        [Column(TypeName = "tinyint")]
        public int ItemGroupId { get; set; }

        [Column(TypeName = "nvarchar(255)")]
        public string SerialNumber { get; set; }

        [Column(TypeName = "nvarchar(255)")]
        public string ArchiveNote { get; set; }

        public List<Archive_StatusHistory>? StatusHistories { get; set; }

    }
}

File: Frontend/src/app/archive/archive-models/archive-item.ts
export interface ArchiveItem {
  id: number;
  roomId: number;
  itemGroupId: number;
  serialNumber: string;
  deleteTime: Date;
  archiveNote: string;
  statusHistories: ArchiveStatusHistory[];
}

export interface ArchiveStatusHistory {
  id: number;
  itemId: number;
  statusId: number;
  statusUpdateDate: Date;
  note: string;
}
